pkg/cmd/cli/secrets/add_secret_to_obj.go

```diff
@@ -42,11 +47,36 @@ var (
 	MountType SecretType = "mount"
 )
 
+type SecretTypeList []SecretType
+
+func (stl *SecretTypeList) String() string {
+	return fmt.Sprint(*stl)
+}
+
+func (stl *SecretTypeList) Set(value string) error {
+	for _, substring := range strings.Split(value, ",") {
+		loweredValue := strings.ToLower(substring)
+		switch loweredValue {
+		case string(PullType):
+			*stl = append(*stl, PullType)
+		case string(MountType):
+			*stl = append(*stl, MountType)
+		default:
+			return fmt.Errorf("unknown for: %v", loweredValue)
+		}
+	}
+	return nil
+}
+
+func (*SecretTypeList) Type() string {
+	return "secretTypeList"
+}
+
 type AddSecretOptions struct {
 	TargetName  string
 	SecretNames []string
 
-	Type SecretType
+	TypeList SecretTypeList
```

Comment:
not sure we want a list here... feels like this is cleaner:
```
AddPullSecret bool
AddMountSecret bool
```

Then you have a single `AddSecretsToSA` function that modifies the SA's list a single time